Assets/Project/Scripts/UIComponents/StatusBarController.cs

```diff
@@ -37,5 +42,38 @@ private static void RunOnUiThread()
             }
             #endif
         }
+
+        /// <summary>
+        /// ステータスバーの下側のアンカーを取得する
+        /// </summary>
+        /// <returns> ステータスバーの高さを取得する</returns>
+        public static float GetStatusBarBottomAnchor()
+        {
+            #if UNITY_IOS || UNITY_EDITOR
+            var safeArea = Screen.safeArea;
+            if (Screen.height == Screen.safeArea.height) {
+                // iPhone X以前の機種
+                var deviceGeneration = UnityEngine.iOS.Device.generation;
+                if (deviceGeneration == UnityEngine.iOS.DeviceGeneration.iPhone6Plus || deviceGeneration == UnityEngine.iOS.DeviceGeneration.iPhone6SPlus || deviceGeneration == UnityEngine.iOS.DeviceGeneration.iPhone7Plus || deviceGeneration == UnityEngine.iOS.DeviceGeneration.iPhone8Plus) {
```

Comment:
```suggestion
                if (deviceGeneration == DeviceGeneration.iPhone6Plus ||
                    deviceGeneration == DeviceGeneration.iPhone6SPlus ||
                    deviceGeneration == DeviceGeneration.iPhone7Plus ||
                    deviceGeneration == DeviceGeneration.iPhone8Plus) {
```

`UnityEngine.iOS` は `using` してるからいらないかと思う。あと、こんな感じに並べた方が読みやすいかな。